pkg/kubectl/cmd/resource/get.go

```diff
@@ -320,40 +371,24 @@ func (options *GetOptions) Run(f cmdutil.Factory, cmd *cobra.Command, args []str
 		objs[ix] = infos[ix].Object
 	}
 
-	sorting, err := cmd.Flags().GetString("sort-by")
-	if err != nil {
-		return err
-	}
+	// sort all objects
 	var sorter *kubectl.RuntimeSort
-	if options.Sort && len(objs) > 1 {
+	if len(options.SortBy) > 0 && len(objs) > 1 {
 		// TODO: questionable
-		if sorter, err = kubectl.SortObjects(cmdutil.InternalVersionDecoder(), objs, sorting); err != nil {
+		if sorter, err = kubectl.SortObjects(cmdutil.InternalVersionDecoder(), objs, options.SortBy); err != nil {
 			return err
 		}
 	}
 
-	// use the default printer for each object
-	printer = nil
-	var lastMapping *meta.RESTMapping
-	w := printers.GetNewTabWriter(options.Out)
+	options.ShowKind = options.ShowKind || resource.MultipleTypesRequested(args) || cmdutil.MustPrintWithKinds(objs, infos, sorter)
```

Comment:
@deads2k 
> This looks wrong. Mutation like this seems like it belongs in complete.

Because this depends on `cmdutil.MustPrintWithKinds`, and that method depends on sorted infos, I would have to move the resource builder logic, as well as sorting to the Complete method, which I believe would have to be gated in the event of `--watch` (or reconciled with the resource builder logic from the `options.watch` method)

Reply:
No, `MustPrintWithKinds` is logically similar to `MultipleTypesRequested` in that it verifies if the returned types are different. Can you think of an example where the requested type will be a single type but server will return multiple different? I can't off top of my head, so I'd limit this to the two conditions and move it to `Complete`.